openfermioncirq/testing/example_classes.py

```diff
@@ -0,0 +1,104 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Subclasses of abstract classes for use in tests."""
+
+from typing import Optional, Sequence, Union, cast
+
+import numpy
+
+import cirq
+
+from openfermioncirq.optimization.algorithm import OptimizationAlgorithm
+from openfermioncirq.optimization.black_box import BlackBox
+from openfermioncirq.optimization.result import OptimizationResult
+from openfermioncirq.variational.ansatz import VariationalAnsatz
+from openfermioncirq.variational.study import VariationalStudy
+
+
+class ExampleAlgorithm(OptimizationAlgorithm):
+
+    def optimize(self,
+                 black_box: BlackBox,
+                 initial_guess: Optional[numpy.ndarray]=None,
+                 initial_guess_array: Optional[numpy.ndarray]=None
+                 ) -> OptimizationResult:
+        numpy.random.seed(13274)
```

Comment:
It may be very surprising to seed the class from within the method like these, because this is a global effect that may kick back into the number generated by the calling method. You should use an instance of the random number generator class and seed that, instead of touching the global seed.